app/src/main/java/com/codesoom/assignment/repository/TaskRepository.java

```diff
@@ -0,0 +1,60 @@
+package com.codesoom.assignment.repository;
+
+import com.codesoom.assignment.models.Task;
+import com.codesoom.assignment.models.TaskIdGenerator;
+import org.springframework.stereotype.Repository;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Repository
+public class TaskRepository {
+
+    private final TaskIdGenerator taskIdGenerator;
+    private Map<Long, Task> taskMap = new ConcurrentHashMap<>();
+
+    public TaskRepository(TaskIdGenerator taskIdGenerator) {
+        this.taskIdGenerator = taskIdGenerator;
+    }
+
+    public Collection<Task> getTaskList() {
+        return taskMap.values();
+    }
+
+    public Optional<Task> getTaskById(Long id) {
+        return Optional.ofNullable(taskMap.get(id));
+    }
+
+    public Task addTask(Task task) {
+        Task newTask = new Task(taskIdGenerator.getNextSequence());
+        newTask.setTitle(task.getTitle());
+        taskMap.put(newTask.getId(), newTask);
+        return newTask;
+    }
+
+    public Optional<Task> replaceTask(Long id, Task task) {
+        Optional<Task> oldTask = Optional.ofNullable(taskMap.get(id));
+        oldTask.ifPresent(thisTask -> {
+            thisTask.setTitle(task.getTitle());
+            taskMap.put(thisTask.getId(), thisTask);
+        });
+        return oldTask;
+    }
```

Comment:
`task`라는 말이 너무 많아서 헷갈리네요. `task, oldTask, thisTask` 너무 많습니다.
변수의 수명이 짧을 경우 단순한 이름으로 사용하는게 오히려 이해하는데 쉬울 수 있습니다.

```suggestion
    public Optional<Task> replaceTask(Long id, Task task) {
        Optional<Task> oldTask = Optional.ofNullable(taskMap.get(id));
        oldTask.ifPresent(it -> {
            it.setTitle(task.getTitle());
            taskMap.put(thisTask.getId(), it);
        });
        return oldTask;
    }
```

`i`를 쓰셔도 됩니다. 저는 코틀린에서 기본으로 `it`를 쓰길래 그냥 `it`를 자주 씁니다

Reply:
간단하게 줄여 써도 가독성이 더 좋네요! 수정했습니다